File: VBNetCore.Cryptography/EncryptionManager.vb
Imports System
Imports System.Security.Cryptography
Imports System.Text

Namespace VBNetCore.Cryptography
    Public Class EncryptionManager
        Public Const KEY As String = "@bcdefghijklmnopqrstuvwxyz1234567890#+-="
        Private Shared _instance As EncryptionManager = Nothing
        Private Sub New()
        End Sub
        Public Shared ReadOnly Property Instance As EncryptionManager
            Get
                If _instance Is Nothing Then _instance = New EncryptionManager()
                Return _instance
            End Get
        End Property
        Public Function PasswordVerify(ByVal hashedPassword As String, ByVal plainPassword As String) As Boolean
            Dim a = EncryptString(plainPassword, KEY)
            Return hashedPassword.Equals(a)
        End Function

        Public Function EncryptString(ByVal Message As String, ByVal Passphrase As String) As String
            Dim uTF8Encoding As UTF8Encoding = New UTF8Encoding()
            Dim md5 = Security.Cryptography.MD5.Create()
            Dim key = md5.ComputeHash(uTF8Encoding.GetBytes(Passphrase))
            Dim tripelDES = TripleDES.Create()
            tripelDES.Key = key
            tripelDES.Mode = CipherMode.ECB
            tripelDES.Padding = PaddingMode.PKCS7
            Dim bytes = uTF8Encoding.GetBytes(Message)
            Dim inArray As Byte()
            Try
                Dim cryptoTransform As ICryptoTransform = tripelDES.CreateEncryptor()
                inArray = cryptoTransform.TransformFinalBlock(bytes, 0, bytes.Length)
            Finally
                tripelDES.Clear()
                tripelDES.Clear()
            End Try
            Return Convert.ToBase64String(inArray)
        End Function
        Public Function DecryptString(ByVal Message As String, ByVal Passphrase As String) As String
            Dim uTF8Encoding As UTF8Encoding = New UTF8Encoding()
            Dim md5 = Security.Cryptography.MD5.Create()
            Dim key = md5.ComputeHash(uTF8Encoding.GetBytes(Passphrase))
            Dim tripleDES = Security.Cryptography.TripleDES.Create()
            tripleDES.Key = key
            tripleDES.Mode = CipherMode.ECB
            tripleDES.Padding = PaddingMode.PKCS7
            Dim array = Convert.FromBase64String(Message)
            Dim bytes As Byte()
            Try
                Dim cryptoTransform As ICryptoTransform = tripleDES.CreateDecryptor()
                bytes = cryptoTransform.TransformFinalBlock(array, 0, array.Length)
            Finally
                tripleDES.Clear()
                tripleDES.Clear()
            End Try
            Return uTF8Encoding.GetString(bytes)
        End Function
    End Class
End Namespace

File: VBNetCore.Cryptography/EncryptionManager.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace NetCore.Cryptography
{
    public class EncryptionManager
    {
        public const string KEY = "@bcdefghijklmnopqrstuvwxyz1234567890#+-=";
        private static EncryptionManager? _instance;
        private EncryptionManager()
        { }
        public static EncryptionManager Instance
        {
            get
            {
                if (_instance == null)
                    _instance = new EncryptionManager();
                return _instance;
            }
        }
        public bool PasswordVerify(string hashedPassword, string plainPassword)
        {
            var a = EncryptString(plainPassword, KEY);
            return hashedPassword.Equals(a);
        }

        public string EncryptString(string Message, string Passphrase)
        {
            UTF8Encoding uTF8Encoding = new UTF8Encoding();
            var md5 = MD5.Create();
            byte[] key = md5.ComputeHash(uTF8Encoding.GetBytes(Passphrase));
            var tripelDES = TripleDES.Create();
            tripelDES.Key = key;
            tripelDES.Mode = CipherMode.ECB;
            tripelDES.Padding = PaddingMode.PKCS7;
            byte[] bytes = uTF8Encoding.GetBytes(Message);
            byte[] inArray;
            try
            {
                ICryptoTransform cryptoTransform = tripelDES.CreateEncryptor();
                inArray = cryptoTransform.TransformFinalBlock(bytes, 0, bytes.Length);
            }
            finally
            {
                tripelDES.Clear();
                tripelDES.Clear();
            }
            return Convert.ToBase64String(inArray);
        }
        public string DecryptString(string Message, string Passphrase)
        {
            UTF8Encoding uTF8Encoding = new UTF8Encoding();
            var md5 = MD5.Create();
            byte[] key = md5.ComputeHash(uTF8Encoding.GetBytes(Passphrase));
            var tripleDES = TripleDES.Create();
            tripleDES.Key = key;
            tripleDES.Mode = CipherMode.ECB;
            tripleDES.Padding = PaddingMode.PKCS7;
            byte[] array = Convert.FromBase64String(Message);
            byte[] bytes;
            try
            {
                ICryptoTransform cryptoTransform = tripleDES.CreateDecryptor();
                bytes = cryptoTransform.TransformFinalBlock(array, 0, array.Length);
            }
            finally
            {
                tripleDES.Clear();
                tripleDES.Clear();
            }
            return uTF8Encoding.GetString(bytes);
        }
    }
}
